src/Achievements/Achievement.cs

```diff
@@ -19,14 +21,42 @@ public class Achievement : UniformTokenSplitter
 
 		protected Address Exchanger { get; set; }
 
+		protected string Describtion { get; set; }
+
+		protected IDictionary<Address, ITaggedTokens> TokenContributors { get; set; }
+			= new Dictionary<Address, ITaggedTokens>();
+
+		protected override void HandleMessage(Message message)
+		{
+			switch (message.Type)
+			{
+				case TokensReceivedEvent.Type:
+					var sender = message.Payload.Get<Address>(TokensReceivedEvent.From);
+					var tokens = message.Payload.GetDictionary(TokensReceivedEvent.TokensTotal);
+					if(sender != null)
+					{
+						if(this.TokenContributors.ContainsKey(sender)) this.TokenContributors[sender].AddToBalance(new TaggedTokens(tokens));
+						else this.TokenContributors.Add(sender, new TaggedTokens(tokens));
+					}
+						
+					this.Split(message.Sender, new ReadOnlyTaggedTokens(tokens));
+					break;
+				default:
+					base.HandleMessage(message);
+					return;
```

Comment:
Unnecessary return statement

Reply:
Nope. It won't compile without a `break` or `return` there.